### PR TITLE
feat(cardano): add support for script data hash in tx

### PR DESCRIPTION
**Pull request by davidmisiak**

### CODE REVIEW

File: common/tests/fixtures/cardano/sign_tx_stake_pool_registration.json [hunk 1]
```diff
@@ -486,6 +492,98 @@
           }
         ]
       }
+    },
+    {
+      "description": "Sample stake pool registration certificate with script data hash",
```
Comment (gabrielKerekes) [inline common/tests/fixtures/cardano/sign_tx_stake_pool_registration.json:497]: I'm not 100% sure this should/will be allowed, but let's leave it for now.

File: core/src/apps/cardano/layout.py [hunk 2]
```diff
@@ -425,6 +425,16 @@ async def confirm_catalyst_registration(
     )
 
 
+async def confirm_script_data_hash(ctx: wire.Context, script_data_hash: bytes) -> None:
+    await confirm_properties(
+        ctx,
+        "confirm_script_data_hash",
+        title="Confirm transaction",
+        props=[("Script data hash:", script_data_hash)],
```
Comment (gabrielKerekes) [inline core/src/apps/cardano/layout.py:433]: Please add a TODO comment here so we don't forget to decide if the hash shouldn't be shown in bech32.

File: core/src/apps/cardano/sign_tx.py [hunk 3]
```diff
@@ -778,6 +799,11 @@ def _validate_witness(
         _ensure_no_payment_witness(witness)
 
 
+def _validate_script_data_hash(script_data_hash: bytes) -> None:
```
Comment (gabrielKerekes) [inline core/src/apps/cardano/sign_tx.py:802]: Please add a TODO comment at the end of this function which would remind us to re-assess the validation once more - specifically with regards to signing modes.

**Review by gabrielKerekes — APPROVED** [review]
Looks good 👍  Just add the TODOs I mentioned please. We'll see what we'll do with this PR when Alonzo work kicks off - until then, let's not do anything with this PR.

At some point UI test fixtures will need to be generated, but that can be done as part of the larger PR in the future.

**Comment by davidmisiak** [comment]
Closing this in favor of https://github.com/vacuumlabs/trezor-firmware/pull/71

**Pull request closed without merging**